src/main/java/com/epam/aidial/core/controller/ResourceOperationController.java

```diff
@@ -76,29 +76,28 @@ public Future<?> move() {
                         throw new IllegalArgumentException("destinationUrl must be provided");
                     }
 
-                    String bucketLocation = BlobStorageUtil.buildInitiatorBucket(context);
-                    String bucket = encryptionService.encrypt(bucketLocation);
+                    ResourceDescription source = ResourceDescription.fromAnyUrl(sourceUrl, encryptionService);
+                    ResourceDescription destination = ResourceDescription.fromAnyUrl(destinationUrl, encryptionService);
 
-                    ResourceDescription sourceResource = ResourceDescription.fromPrivateUrl(sourceUrl, encryptionService);
-                    if (!sourceResource.getBucketName().equals(bucket)) {
-                        throw new IllegalArgumentException("sourceUrl do not belong to the user");
+                    if (!source.getType().equals(destination.getType())) {
+                        throw new IllegalArgumentException("source and destination resources must be the same type");
                     }
 
-                    ResourceDescription destinationResource = ResourceDescription.fromPrivateUrl(destinationUrl, encryptionService);
-                    if (!destinationResource.getBucketName().equals(bucket)) {
-                        throw new IllegalArgumentException("destinationUrl do not belong to the user");
+                    if (source.getUrl().equals(destination.getUrl())) {
+                        throw new IllegalArgumentException("source and destination resources cannot be the same");
                     }
 
-                    if (!sourceResource.getType().equals(destinationResource.getType())) {
-                        throw new IllegalArgumentException("source and destination resources must be the same type");
-                    }
+                    Set<ResourceDescription> resources = Set.of(source, destination);
```

Comment:
The permission check is not quite correct. A requester should have read access to source and write access to destination.

Reply:
Move removes the source resource (not just reads)

Reply:
Do you want me to check that there is read/write access to source and just write access to destination?

Reply:
You're right. Ideally a source should have both Read and Write access and a destination is just a write.